MatrixSDK/Background/Store/Model/MXCachedSyncResponse.m

```diff
@@ -14,31 +14,46 @@
 // limitations under the License.
 //
 
-#import "MXSyncResponseStoreModel.h"
+#import "MXCachedSyncResponse.h"
 #import "MXJSONModels.h"
 
-@implementation MXSyncResponseStoreModel
+@implementation MXCachedSyncResponse
+
+- (instancetype)initWithSyncToken:(NSString *)syncToken syncResponse:(MXSyncResponse *)syncResponse
+{
+    self = [super init];
+    if (self)
+    {
+        _syncToken = syncToken;
+        _syncResponse = syncResponse;
+    }
+    return self;
+}
 
 + (instancetype)modelFromJSON:(NSDictionary *)JSONDictionary
 {
-    MXSyncResponseStoreModel *syncResponseModel = [[MXSyncResponseStoreModel alloc] init];
+    MXCachedSyncResponse *cachedSyncResponse;
+    
+    NSString *syncToken;
+    MXSyncResponse *syncResponse;
+    MXJSONModelSetString(syncToken, JSONDictionary[@"sync_token"]);
```

Comment:
`MXSyncResponseFileStore` has a new file organisation. We will not decode an old file that used `prev_batch`. The code will not crash because of this.

I do not want to manage a migration because `MXSyncResponseFileStore` is just a skippable cache for the moment.

What is true is that we do not erase previous data. I am going to fix it.

Reply:
Makes sense, thanks.